README.md

```diff
@@ -1,35 +1,351 @@
+- UNLIKELY
+- VERY_UNLIKELY
+
+### Detector
+
+A detector is either a prebuilt detector from nightfall or customized regex | wordlist by customer, specified by
+detectorType field.
+
+- nightfall prebuilt detector
+
+  ```json
+  {
+    "detector": {
+      "detectorType": "NIGHTFALL_DETECTOR",
+      "nightfallDetector": "API_KEY",
+      "displayName": "credit card detector"
+    }
+  }
+  ```
+
+  Within detector struct
+
+  - First specify detectorType as NIGHTFALL_DETECTOR
+  - Pick the nightfall detector you want from the list
+    - API_KEY
+    - CRYPTOGRAPHIC_KEY
+    - RANDOMLY_GENERATED_TOKEN
+    - CREDIT_CARD_NUMBER
+    - US_SOCIAL_SECURITY_NUMBER
+    - AMERICAN_BANKERS_CUSIP_ID
+    - US_BANK_ROUTING_MICR
+    - ICD9_CODE
+    - ICD10_CODE
+    - US_DRIVERS_LICENSE_NUMBER
+    - US_PASSPORT
+    - PHONE_NUMBER
+    - IP_ADDRESS
+    - EMAIL_ADDRESS
+  - Put a display name for your detector, as this will be attached on your findings
+
+- customized regex
+
+  For convenience, we also support customized regex as a detector, which you can fill in as follow.
+
+  ```json
+  {
+    "detector": {
+      "detectorType": "REGEX",
+      "regex": {
+        "pattern": "coupon",
+        "isCaseSensitive": false
+      },
+      "displayName": "simpleRegexCouponDetector"
+    }
+  }
+  ```
+
+- word list
+
+  In case you are not famaliar what regex is, we also support word list searching, simply put in a list of words you want to find.
+
+  ```json
+  {
+    "detector": {
+      "detectorType": "WORD_LIST",
+      "wordList": {
+        "values": ["key", "credential"],
+        "isCaseSensitive": false
+      },
+      "displayName": "simpleWordListKeyDetector"
+    }
+  }
+  ```
+
+- [Extra Parameters Within Detector]
+
+  Besides to define which detector to call, we also allow you to optionally specify some rules you want to attach surround the findinds and finding themselves. They are contextRules and exclusionRules.
+
+  - contextRules
+    A context rule is defined as the surrounding context(pre/post chars) of a finding, you can define a trigger/rule in this section. Once it's triggered, the confidence of finding will be adjusted accordingly base on your confid.
+
+    Example:
+
+    ```json
+    {
+      "detector": {
+        // ...... other detector fileds
+        "contextRules": [
+          {
+            "regex": {
+              "pattern": "test cc",
+              "isCaseSensitive": true
+            },
+            "proximity": {
+              "windowBefore": 30,
+              "windowAfter": 30
+            },
+            "confidenceAdjustment": {
+              "fixedConfidence": "VERY_UNLIKELY"
+            }
+          }
+        ]
+      }
+    }
+    ```
+
+    - regex field specifies a regex to trigger
+    - proximity specifies how many pre|post chars surround findings we want to do the search
+    - confidenceAdjustment specifies what's the confidence you want to change for the finding once trigger the rule
```

Comment:
I think we can get rid of regex since we already explained that above for the regex detector.
```suggestion
    - proximity is defined as the number pre|post chars surrounding the finding to conduct the search
    - confidenceAdjustment is the confidence level to adjust the finding to upon existence of the input context
```

Reply:
not sure if the reader would read everything from top to bottom,
I think it's reasonable to assume he/she only reads a block of content, thus kept the regex section

changed others